File: DecompiledGameScripts/core/gameplay/effectors/broadcastStimEffector.swift

public class BroadcastStimEffector extends ContinuousEffector {

  public let m_stimType: gamedataStimType;

  public let m_radius: Float;

  protected func Initialize(record: TweakDBID, game: GameInstance, parentRecord: TweakDBID) -> Void {
    let effectorRecord: ref<BroadcastStimEffector_Record> = TweakDBInterface.GetBroadcastStimEffectorRecord(record);
    if IsDefined(effectorRecord) {
      this.m_stimType = effectorRecord.Type().Type();
      this.m_radius = effectorRecord.Radius();
    };
  }

  protected func ContinuousAction(owner: ref<GameObject>, instigator: ref<GameObject>) -> Void {
    StimBroadcasterComponent.BroadcastStim(owner, this.m_stimType, this.m_radius);
  }
}

File: Definitions/Game_Definitions/classes/BroadcastStimEffector.lua
---@meta
---@diagnostic disable

---@class BroadcastStimEffector : gameContinuousEffector
---@field stimType gamedataStimType
---@field radius Float
BroadcastStimEffector = {}

---@return BroadcastStimEffector
function BroadcastStimEffector.new() return end

---@param props table
---@return BroadcastStimEffector
function BroadcastStimEffector.new(props) return end

---@param owner gameObject
---@param instigator gameObject
function BroadcastStimEffector:ContinuousAction(owner, instigator) return end

---@param record TweakDBID|string
---@param parentRecord TweakDBID|string
function BroadcastStimEffector:Initialize(record, parentRecord) return end

